### ISSUE TITLE
Update documentation to include templated MxArray conversions

### ISSUE THREAD

**Issue by dnmiller** [kyamagu/mexplus#25]
I'd like to first say thank you for this library. It's very useful.

I was trying to extend MxArray with conversions to/from std::array and was running into problems with deducing the type/size of the array. However, I was able to  add the conversion functions by simply not making them members of the MxArray class. For example, the following code works:

```c++
namespace mexplus
{
    template <typename T, size_t N>
    mxArray* from(const std::array<T, N>& value) {
        MxArray result(MxArray::Numeric<T>(N, 1));
        result.set(value);
        return result.release();
    }

    template <typename T,  size_t N>
    void to(const mxArray* array, std::array<T, N>* value) {
        MxArray input(array);
        std::copy(value.begin(), value.end(),
                  reinterpret_cast<T*>(mxGetData(array)));
    }
}
```

However, if I change `to` and `from` to `MxArray::to` and `MxArray::from`, then it won't compile.

After thinking about it some more, this may be a preferable way to defining conversions. It would be nice to at least have it in the documentation.

**Comment by kyamagu** [comment]
@dnmiller Hmm, you are right. Templated conversion might not work under current design. This is because there are a bunch of type inference for container template inside, and `std::array` will probably fall into a container kind. I don't know if documentation update is enough here.

**Comment by dnmiller** [comment]
Yes, that appears to be what happened. I might try and work on this if I get some spare time. Any idea what a good approach might be? Make the type inference for fromInternal more restrictive?

**Comment by kyamagu** [comment]
@dnmiller It would be great if you have a spare time to look into the issue. The current design defines [type traits](https://github.com/kyamagu/mexplus/blob/master/include/mexplus/mxtypes.h) to instantiate [various conversions](https://github.com/kyamagu/mexplus/blob/master/include/mexplus/mxarray.h) based on type. It is probably easy to treat `std::array` just like `std::vector`. Making a template conversion on top of MxArray methods may need something more.

**Comment by patrikhuber** [comment]
I would also be very much interested in this!
In fact I am just implementing `to` & `from` conversions for `Eigen::MatrixXd`. Actually, reading this issue, I am a bit surprised that so far it has worked for me, since `MatrixXd` is an alias to the templated class `Matrix<typename _Scalar, int _Rows, int _Cols, int _Options, int _MaxRows, int _MaxCols>`. But it's certainly possible that I'll hit a roadblock soon. Is there any update from either of you?

**Comment by patrikhuber** [comment]
Ah, I think I get it - in my case all template parameters are defined, so I am not hitting this problem, because I'm not templating my to/from methods. However, if I wanted to write a more generic conversion, then I'd have to "templatise" to/from and then I would hit this problem as well.
So yes I would very much be interested in this! :-)
